taichi/ir/type_utils.h

```diff
@@ -124,5 +124,57 @@ inline bool needs_grad(DataType dt) {
   return is_real(dt);
 }
 
+inline TypedConstant get_max_value(DataType dt) {
+  if (dt->is_primitive(PrimitiveTypeID::i8)) {
```

Comment:
nit: in the future maybe we can predefine something like a `PER_TYPE(i8, int8)`, then we don't have to repeat here..